kafka-plugins-0.10/src/main/java/io/cdap/plugin/source/KafkaStreamingSource.java

```diff
@@ -62,13 +65,25 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
     }
   }
 
+  @Override
+  public void prepareRun(StreamingSourceContext context) throws Exception {
+    Schema schema = conf.getSchema(context.getFailureCollector());
+    // record dataset lineage
+    context.registerLineage(conf.referenceName, schema);
```

Comment:
why do we need to register lineage on context vs in batch we do not call any method on context directly.
https://github.com/data-integrations/google-cloud/blob/develop/src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java#L429

Reply:
Since we cannot make the external datasets trackable in streaming pipeline due to spark streaming, so we have to register the usage of the external datasets ourselves. And it has to happen in prepareRun since the getStream is not getting called each time the pipeline run.